### ISSUE TITLE
math.format() with {notation: 'fixed'} does not work for expressions

### ISSUE THREAD

**Issue by dkreye** [josdejong/mathjs#1606]
Bug found in version 6.1.0:
The function `math.format` does not propagate its argument `options` into the recursive subcalls. As a consequence, the options are ignored if applied to expressions.

Example:
`math.format( math.parse( "1234567 + 1"), {notation: 'fixed'})`
results in "1,234567 + 1" rather than the expected "1234567 + 1".

**Comment by dkreye** [comment]
In order to fix this bug, propagating `options` into the `toString()` calls in function `string_format` is a good start:

```
@@ -2540,9 +2540,9 @@
   if (value && _typeof(value) === 'object') {
     if (typeof value.format === 'function') {
       return value.format(options);
-    } else if (value && value.toString(options) !== {}.toString()) {
+    } else if (value && value.toString() !== {}.toString()) {
       // this object has a non-native toString method, use that one
-      return value.toString(options);
+      return value.toString();
     } else {
       var entries = Object.keys(value).map(function (key) {
         return '"' + key + '": ' + string_format(value[key], options);

```

**Comment by dkreye** [comment]
Workaround: `math.parse( "1234567 + 1").toString( {notation: 'fixed'})`

**Comment by josdejong** [comment]
Thanks @dkreye, fixed via e37cc81 (not yet published).

It remains a bit odd that we pass `options` to the default `.toString`, maybe some day we can better rename this method to `.format(options)` since it's non-standard behavior (`.toString` has no arguments).

**Comment by josdejong** [comment]
Fixed now in `v6.2.1`

**Comment by dkreye** [comment]
Thanks a lot, that was quick...
